src/biconnectivity/articulation.jl

```diff
@@ -22,74 +23,136 @@ julia> articulation(path_graph(5))
 function articulation end
 @traitfn function articulation(g::AG::(!IsDirected)) where {T,AG<:AbstractGraph{T}}
     s = Vector{Tuple{T,T,T}}()
-    is_articulation_pt = falses(nv(g))
     low = zeros(T, nv(g))
     pre = zeros(T, nv(g))
 
+    is_articulation_pt = falses(nv(g))
     @inbounds for u in vertices(g)
-        pre[u] != 0 && continue
-        v = u
-        children = 0
-        wi::T = zero(T)
-        w::T = zero(T)
-        cnt::T = one(T)
-        first_time = true
-
-        # TODO the algorithm currently relies on the assumption that
-        # outneighbors(g, v) is indexable. This assumption might not be true
-        # in general, so in case that outneighbors does not produce a vector
-        # we collect these vertices. This might lead to a large number of
-        # allocations, so we should find a way to handle that case differently,
-        # or require inneighbors, outneighbors and neighbors to always
-        # return indexable collections.
-
-        while !isempty(s) || first_time
-            first_time = false
-            if wi < 1
-                pre[v] = cnt
-                cnt += 1
-                low[v] = pre[v]
-                v_neighbors = collect_if_not_vector(outneighbors(g, v))
-                wi = 1
-            else
-                wi, u, v = pop!(s)
-                v_neighbors = collect_if_not_vector(outneighbors(g, v))
-                w = v_neighbors[wi]
-                low[v] = min(low[v], low[w])
-                if low[w] >= pre[v] && u != v
+        articulation_dfs!(is_articulation_pt, g, u, s, low, pre)
+    end
+
+    articulation_points = T[v for (v, b) in enumerate(is_articulation_pt) if b]
+
+    return articulation_points
+end
+
+"""
+    is_articulation(g, v)
+
+Determine whether `v` is an
+[articulation point](https://en.wikipedia.org/wiki/Biconnected_component) of an undirected
+graph `g`, returning `true` if so and `false` otherwise.
+
+See also [`articulation`].
+
+# Examples
+```jldoctest
+julia> using Graphs
+
+julia> g = path_graph(5)
+{5, 4} undirected simple Int64 graph
+
+julia> articulation(g)
+3-element Vector{Int64}:
+ 2
+ 3
+ 4
+
+julia> is_articulation(g, 2)
+true
+
+julia> is_articulation(g, 1)
+false
+```
+"""
+function is_articulation end
+@traitfn function is_articulation(g::AG::(!IsDirected), v::T) where {T,AG<:AbstractGraph{T}}
+    s = Vector{Tuple{T,T,T}}()
+    low = zeros(T, nv(g))
+    pre = zeros(T, nv(g))
+
+    return articulation_dfs!(nothing, g, v, s, low, pre)
+end
+
+@traitfn function articulation_dfs!(
+    is_articulation_pt::Union{Nothing,BitVector},
+    g::AG::(!IsDirected),
+    u::T,
+    s::Vector{Tuple{T,T,T}},
+    low::Vector{T},
+    pre::Vector{T},
```

Comment:
What do these arguments mean?

Reply:
I admit, I don't recall: this was mainly a refactoring of the original code and I just kept the variable names as they were.

Reply:
fair enough